### DISCUSSION TITLE
Inconsistent Error Code in Modbus Client

### DISCUSSION THREAD

**Discussion by aatoof1** [zephyrproject-rtos/zephyr#78061] [RFC]
Hi, 

In function `mbc_validate_wr_response` line `230`, the return code is `ENXIO` without the minus sign, is this a bug? 
https://github.com/zephyrproject-rtos/zephyr/blob/4677920956ed27a6903fb3a520c20dcf15abd325/subsys/modbus/modbus_client.c#L224-L234

In the rest of the code if a positive return code is returned, this indicates a Modbus exception code. See:
https://github.com/zephyrproject-rtos/zephyr/blob/4677920956ed27a6903fb3a520c20dcf15abd325/subsys/modbus/modbus_client.c#L46-L52

Also in other parts of the code when there is a mismatch between request and response values (i.e function code, unit id) the error code `-EIO` is returned, with the above case being the only exception. See:
https://github.com/zephyrproject-rtos/zephyr/blob/4677920956ed27a6903fb3a520c20dcf15abd325/subsys/modbus/modbus_client.c#L38-L44

Should `ENXIO` be changed to `-EIO` to be consistent with the rest of the Modbus code?

Thanks

**Comment by jukkar** [comment]
@aatoof1 I think you could send a PR and propose this change, then it gets assigned and reviewed by relevant people.
